test/index.test.js

```diff
@@ -170,6 +189,33 @@ describe('plugin', () => {
     });
   });
 
+  it('sets custom rateLimitKey function given', () => {
+    return server.injectThen({
+      method: 'POST',
+      url: '/custom_rate_limit_key_test',
+      credentials: { api_key: '123' }
+    })
+    .then((response) => {
+      expect(response.result.rate.limit).to.eql(defaultRate.limit);
+      expect(response.result.rate.window).to.eql(defaultRate.window);
+      const key = redisSpy.firstCall.args[0];
+      expect(key).to.match(/hapi-rate-limiter:\S+:\S+:custom/);
```

Comment:
I don't think there's a need to set `key` since it's only used in the `expect(...)` function.